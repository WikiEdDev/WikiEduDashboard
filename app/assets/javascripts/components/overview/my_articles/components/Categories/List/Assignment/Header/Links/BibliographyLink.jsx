import React from 'react';
import PropTypes from 'prop-types';

export const BibliographyLink = ({ assignment }) => {
  const url = `${assignment.sandboxUrl}/Bibliography?veaction=edit&preload=Template:Dashboard.wikiedu.org_bibliography`;
  return <a href={url} target="_blank">{I18n.t('assignments.bibliography')}</a>;
};

BibliographyLink.propTypes = {
  // props
  assignment: PropTypes.object.isRequired,
};

export default BibliographyLink;
