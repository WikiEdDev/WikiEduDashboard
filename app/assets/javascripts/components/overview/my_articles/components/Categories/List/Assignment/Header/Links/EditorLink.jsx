import React from 'react';
import PropTypes from 'prop-types';

import AssignedToLink from '../../../../../../common/AssignedToLink';

export const EditorLink = ({ editors }) => {
  return <AssignedToLink members={editors} name="editors" />;
};

EditorLink.propTypes = {
  // props
  editors: PropTypes.array,
};

export default EditorLink;
