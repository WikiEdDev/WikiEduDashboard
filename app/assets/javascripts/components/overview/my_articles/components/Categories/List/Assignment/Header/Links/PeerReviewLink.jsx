import React from 'react';
import PropTypes from 'prop-types';

export const PeerReviewLink = ({ assignment, current_user }) => {
  const { sandboxUrl } = assignment;
  const { username } = current_user;
  let peerReviewUrl = `${sandboxUrl}/${username}_Peer_Review`;
  peerReviewUrl += '?veaction=edit&preload=Template:Dashboard.wikiedu.org_peer_review';
  return <a href={peerReviewUrl} target="_blank">{I18n.t('assignments.peer_review_link')}</a>;
};

PeerReviewLink.propTypes = {
  assignment: PropTypes.object.isRequired,
  current_user: PropTypes.object.isRequired,
};

export default PeerReviewLink;
