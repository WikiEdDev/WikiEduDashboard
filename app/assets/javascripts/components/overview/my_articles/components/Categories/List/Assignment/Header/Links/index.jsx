import React from 'react';
import PropTypes from 'prop-types';

// components
import BibliographyLink from './BibliographyLink';
import SandboxLink from './SandboxLink';
import PeerReviewLink from './PeerReviewLink';
import EditorLink from './EditorLink';
import ReviewerLink from '../../../../../../common/ReviewerLink';

import Separator from '../../../../../../common/Separator';

// constants
import { REVIEWING_ROLE } from '../../../../../../../../../constants/assignments';

// helper functions
const interleaveSeparators = (acc, link, index, collection) => {
  const limit = collection.length - 1;
  const prefix = [...acc, link];

  return index < limit ? prefix.concat(<Separator key={index} />) : prefix;
};

export const Links = ({ articleTitle, assignment, current_user }) => {
  const { article_url, editors, id, reviewers } = assignment;

  let actions = [
    <BibliographyLink key={`bibliography-${id}`} assignment={assignment} />,
    <SandboxLink key={`sandbox-${id}`} assignment={assignment} />
  ];

  if (assignment.role === REVIEWING_ROLE) {
    actions.push(
      <PeerReviewLink key={`review-${id}`} assignment={assignment} current_user={current_user} />
    );
  }

  const article = (
    <a key={`article-${id}`} href={article_url} target="_blank">{I18n.t('assignments.article_link')}</a>
  );

  actions = actions.concat(article).reduce(interleaveSeparators, []);

  const assignedTo = (editors || reviewers)
    ? (
      <>
        <EditorLink key={`editor-${id}`} editors={editors} />
        {(editors && reviewers) ? <Separator key="member-links" /> : null}
        <ReviewerLink key={`reviewer-${id}`} reviewers={reviewers} />
      </>
    )
    : null;

  return (
    <section className="header">
      <section className="title">
        <h4>{articleTitle}</h4>
      </section>
      <section className="editors">
        {assignedTo && <p className="mb0">{assignedTo}</p>}
        <p>{actions}</p>
      </section>
    </section>
  );
};

Links.propTypes = {
  // props
  articleTitle: PropTypes.string.isRequired,
  assignment: PropTypes.object.isRequired,
  current_user: PropTypes.object.isRequired,
};

export default Links;
