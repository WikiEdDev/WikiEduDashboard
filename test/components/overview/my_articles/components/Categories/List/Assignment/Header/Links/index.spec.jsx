import React from 'react';
import { shallow } from 'enzyme';
import '../../../../../../../../../testHelper';

import Links from '../../../../../../../../../../app/assets/javascripts/components/overview/my_articles/components/Categories/List/Assignment/Header/Links';

describe('Links', () => {
  const props = {
    articleTitle: 'title',
    assignment: { id: 1, article_url: 'url' },
    current_user: { id: 99 }
  };

  it('should show the default links', () => {
    const component = shallow(<Links {...props} />);

    expect(component.find('BibliographyLink').length).toEqual(1);
    expect(component.find('SandboxLink').length).toEqual(1);
    expect(component.find('a[href="url"]').length).toEqual(1);
  });

  it('should show the peer review link if the assignment role is set to reviewing', () => {
    const newProps = { ...props, assignment: { ...props.assignment, role: 1 } };
    const component = shallow(<Links {...newProps} />);
    expect(component.find('PeerReviewLink').length).toEqual(1);
  });

  it('should show EditorLink and ReviewerLink if there are editors or reviewers', () => {
    const assignment = {
      ...props.assignment,
      editors: ['editor'],
      reviewers: ['reviewer']
    };
    const component = shallow(<Links {...props} assignment={assignment} />);
    expect(component.find('EditorLink').length).toEqual(1);
    expect(component.find('ReviewerLink').length).toEqual(1);
  });
});
