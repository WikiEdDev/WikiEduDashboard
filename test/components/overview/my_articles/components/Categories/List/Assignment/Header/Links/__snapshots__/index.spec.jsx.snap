// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links should show the default links 1`] = `ShallowWrapper {}`;
